arch/riscv/core/isr.S

```diff
@@ -375,6 +389,9 @@ no_fp:	/* increment _current->arch.exception_depth */
 	 * no_reschedule to restore stack.
 	 */
 	mv a0, sp
+#ifdef CONFIG_EXCEPTION_DEBUG
+	STORE_CSF_IN_ESF(a0)		;
```

Comment:
@npitre I'm trying to dump the csf on errors like 'illegal instructions' that's going to be handled by `_Fault`, but this causes 'Instruction Access fault' when the error came from the userspace, any idea why?

Reply:
I probably corrupted the `sp` here, if `_Fault` returns?

Reply:
Should be fixed, nvm